test/AspNetCoreModule.Test/Framework/InitializeTestMachine.cs

```diff
@@ -65,27 +83,135 @@ public class InitializeTestMachine : IDisposable
+                    {
+                        _globalTestFlags = ANCMTestFlags_DefaultContext_FullIIS;
+                    }
+
+                    // adjust the default test context in run time to figure out wrong test context values
+                    if (!isElevated)
+                    {
+                        if (_globalTestFlags.Contains(ANCMTestFlags.UseFullIISContext.ToLower()))
+                        {
+                            _globalTestFlags = _globalTestFlags.Replace(ANCMTestFlags.UseFullIISContext.ToLower(), "");
+                        }
+
+                        if (!_globalTestFlags.Contains(ANCMTestFlags.UseIISExpressContext.ToLower()))
+                        {
+                            TestUtility.LogInformation("Added test context of " + ANCMTestFlags.UseIISExpressContext);
+                            _globalTestFlags += ";" + ANCMTestFlags.UseIISExpressContext;
+                        }
+
+                        if (!_globalTestFlags.Contains(ANCMTestFlags.RunAsNonAdminUser.ToLower()))
+                        {
+                            TestUtility.LogInformation("Added test context of " + ANCMTestFlags.RunAsNonAdminUser);
+                            _globalTestFlags += ";" + ANCMTestFlags.RunAsNonAdminUser;
+                        }
+                    }
+
+                    if (MakeCertExeAvailable == true)
+                    {
+                        if (!_globalTestFlags.Contains(ANCMTestFlags.MakeCertExeAvailable.ToLower()))
+                        {
+                            TestUtility.LogInformation("Added test context of " + ANCMTestFlags.MakeCertExeAvailable);
+                            _globalTestFlags += ";" + ANCMTestFlags.MakeCertExeAvailable;
+                        }
+                    }
+
+                    if (!Environment.Is64BitOperatingSystem)
+                    {
+                        if (!_globalTestFlags.Contains(ANCMTestFlags.X86Platform.ToLower()))
+                        {
+                            TestUtility.LogInformation("Added test context of " + ANCMTestFlags.X86Platform);
+                            _globalTestFlags += ";" + ANCMTestFlags.X86Platform;
+                        }
+                    }
+
+                    if (Environment.Is64BitOperatingSystem && !Environment.Is64BitProcess)
+                    {
+                        if (!_globalTestFlags.Contains(ANCMTestFlags.Wow64BitMode.ToLower()))
+                        {
+                            TestUtility.LogInformation("Added test context of " + ANCMTestFlags.Wow64BitMode);
+                            _globalTestFlags += ";" + ANCMTestFlags.Wow64BitMode;
+                        }
+                    }
+
+                    _globalTestFlags = _globalTestFlags.ToLower();
+                }
+
+                return _globalTestFlags;
+            }            
+        }
+
         public InitializeTestMachine()
         {
             _referenceCount++;
```

Comment:
Can you break up this method into multiple methods? I'm having trouble understanding what it is doing.

Reply:
Fixed